### ISSUE TITLE
Inconsistent percent-encoding of keys

### ISSUE THREAD

**Issue by rthill91** [arangodb/arangodb#13187]
## My Environment

* __ArangoDB Version__:        3.7.3
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               Default
* __Infrastructure__:               own
* __Operating System__:        Ubuntu 20.04
* __Total RAM in your machine__:        32Gb
* __Disks in use__:                  SSD
* __Used Package__:              Docker - official Docker library

## Component, Query & Data

__Affected feature__:
AQL query using web interface

__AQL query (if applicable)__:
```
insert
{
"_key": "test_%25"
}
into collection_name

```

## Steps to reproduce

1. Insert document containing percent-encodable key (e.g. `test_%25`)
2. Attempt retrieval of the document in collection view or through rest api

__Problem__:
There is inconsistent percent-encoding happening. Documents are inserted correctly as given, but on retrieval are decoded.

I first noticed this problem when searching for a document by key in the collections view. The document was found, but when I click to open it up I get a DocumentNotFoundError. Manually fixing the key encoding in the URL results in the expected document (e.g. `.../collection/collection_name/test_%2525`)

__Expected result__:
Encoding (or lack-thereof) is consistent across insert and retrieval

**Comment by goedderz** [comment]
Hi @rthill91,

that sound's correct to me. Percent-encoding (also known as URL encoding) is necessary only in the URL.

If you disagree, please describe in more detail what you're doing, what the result is, and what you find unexpected about the result.

**Comment by rthill91** [comment]
The only aspect of this I think is an arangodb issue is what I mentioned about opening documents from the collection view of the web interface. A document with a key containing a substring that looks like a percent-encoded value will not open correctly.

**Comment by Simran-B** [comment]
```js
RETURN collection_name // [ [ { "_key": "test_%25" } ] ]
```

![image](https://user-images.githubusercontent.com/7819991/101893541-2cab5280-3ba5-11eb-83fd-75d1042982b2.png)

Clicking on the document results in:

![image](https://user-images.githubusercontent.com/7819991/101893615-4351a980-3ba5-11eb-84ab-c4e2d70761e2.png)

Same error if you create an edge with key `%25`. The key is correctly set to `%25` however.

It also displays the name incorrectly, and the _from/_to links are broken (same error as above):

![image](https://user-images.githubusercontent.com/7819991/101894124-00440600-3ba6-11eb-949d-8a90ba06e848.png)

**Comment by Simran-B** [comment]
Below function tries to decode the URL with the key not encoded first, and if it fails, it encodes the key. This test doesn't work because `test%25` decodes to a valid URL `test%`:

_js\apps\system\_admin\aardvark\APP\frontend\js\views\documentsView.js_

```js
    goToDocument: function (error, data, msg) {
      if (error) {
        arangoHelper.arangoError('Error', msg.errorMessage);
      } else {
        window.modalView.hide();
        data = data.split('/');

        var url;
        try {
          url = 'collection/' + data[0] + '/' + data[1];
          decodeURI(url);
        } catch (ex) {
          url = 'collection/' + data[0] + '/' + encodeURIComponent(data[1]);
        }

        window.location.hash = url;
      }
    },
```

I'm not sure why it isn't encoded by default.

**Comment by goedderz** [comment]
> The only aspect of this I think is an arangodb issue is what I mentioned about opening documents from the collection view of the web interface. A document with a key containing a substring that looks like a percent-encoded value will not open correctly.

Ah, I see. That's not right, of course.

> I'm not sure why it isn't encoded by default.

That looks strange indeed, and I'd expect it to be unconditionally encoded, too.

**Comment by Simran-B** [comment]
This one is even weirder:

_js\apps\system\_admin\aardvark\APP\frontend\js\views\documentView.js:_

```js
      var navigateTo = $(e.target).attr('documentLink');
      var test = (navigateTo.split('%').length - 1) % 3;

      if (decodeURIComponent(navigateTo) !== navigateTo && test !== 0) {
        navigateTo = decodeURIComponent(navigateTo);
      }

      if (navigateTo) {
        window.App.navigate(navigateTo, {trigger: true});
      }
```

**Comment by Simran-B** [comment]
Should be fixed as of 3.9.0 by https://github.com/arangodb/arangodb/pull/14700
